### PR TITLE
[SECURITY-3127][SECURITY-3137]

### PR DESCRIPTION
**Pull request by taylor-fuller**
## Description
[IST-610](https://mabl.atlassian.net/browse/IST-610)

This PR: 
- Patches security vulnerabilities related to Jenkins tickets [SECURITY-3127](https://issues.jenkins.io/browse/SECURITY-3127) and [SECURITY-3137](https://issues.jenkins.io/browse/SECURITY-3137)
- Updates minimum Jenkins version to `v2.319.1`

### Testing done
- Updated existing test coverage to accommodate new changes
- Tested e2e locally

### CODE REVIEW

**Review by simonychoy — APPROVED** [review]
Thanks!

File: src/test/java/com/mabl/integration/jenkins/MablStepDescriptorTest.java [hunk 1]
```diff
@@ -7,27 +7,33 @@
...
 
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
     @Before
     public void setup() {
         mablStepDescriptor = mock(MablStepBuilder.MablStepDescriptor.class);
         doNothing().when(mablStepDescriptor).load();
+        // create a dummy security realm
+        j.jenkins.setSecurityRealm(j.createDummySecurityRealm());
```
Comment (twistedpair) [inline src/test/java/com/mabl/integration/jenkins/MablStepDescriptorTest.java:36]: Thanks for tightening this up, @taylor-fuller

File: src/test/java/com/mabl/integration/jenkins/validation/MablBuildValidatorTest.java [hunk 2]
```diff
@@ -32,17 +33,26 @@
...
  */
 public class MablBuildValidatorTest {
+    FreeStyleProject freeStyleProject;
+
+    @Rule
+    public JenkinsRule jenkinsRule = new JenkinsRule();
+
+    @Before
+    public void setupTest() {
+        freeStyleProject = jenkinsRule.jenkins.getItemByFullName("my-freestyle", FreeStyleProject.class);
```
Comment (twistedpair) [inline src/test/java/com/mabl/integration/jenkins/validation/MablBuildValidatorTest.java:43]: Same here 👍

**Review by twistedpair — APPROVED** [review]
Thanks, @taylor-fuller !

**Pull request merged**